pkg/stanza/operator/input/file/config.go

```diff
@@ -38,8 +36,8 @@ type Config struct {
 }
 
 // Build will build a file input operator from the supplied configuration
-func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
-	inputOperator, err := c.InputConfig.Build(logger)
+func (c Config) Build(buildInfo *operator.BuildInfoInternal) (operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(buildInfo.Logger)
```

Comment:
Many operators rely on these composable helper structs. It's hard to say without seeing how we are using the CreateSettings, but I suspect we should pass it through and have the shared structs manage as much of the additional complexity as possible.

Reply:
In my opinion, shared structs do introduce some additional complexity, but they can also facilitate iteration or refactoring. Take the example of adding telemetry, we may need another temporary parameter like "useOtel" If we have the struct, the parameter can easily pass it through without modifying all implementations of the "Builder" interface. Or there will be other parameters that need to be passed through for some new features in the future. Without the struct, all these changes would be very difficult.  How do you think about these situations?

Reply:
I can appreciate the utility of what you are suggesting but I think there is a tradeoff that we have additional complexity which may never be needed. Generally in this repo we strongly prefer to include only what is necessary, even though it occasionally means we need to make a change later.

Reply:
So you mean that we can change the interface like this?

> func (c Config) Build(logger *zap.SugaredLogger, ID component.ID, settings component.TelemetrySettings, telemetryUseOtel bool) (operator.Operator, error)
Does this really reduce much complexity compared to using struct?

Reply:
> So you mean that we can change the interface like this?
> 
> > func (c Config) Build(logger *zap.SugaredLogger, ID component.ID, settings component.TelemetrySettings, telemetryUseOtel bool) (operator.Operator, error)

If in the future we have a need for additional parameters, they will either be required or optional. If they are required, this will be a breaking change anyways. If they are optional (or have reasonable defaults), then there are other ways to handle this beside breaking the signature. We can add variadic options or alternate signatures.

> Does this really reduce much complexity compared to using struct?

Why do functions allow multiple parameters? Obviously there are tradeoffs regarding complexity.

In any case, I will continue to push for minimal complexity in the current implementation. To that end, I think we are two steps ahead of ourselves because we haven't even demonstrated how any of the proposed parameters will actually be used.

Reply:
@djaglowski